includePayload feature (fixes issue #3272) (#3683)
telefonicaid/fiware-orion: doc/manuals/user/ngsiv2_implementation_notes.md

* [Registrations](#registrations)
* [`keyValues` not supported in `POST /v2/op/notify`](#keyvalues-not-supported-in-post-v2opnotify)
* [Deprecated features](#deprecated-features)

> * [Additional fields in `httpInfo` for custom notifications](#additional-fields-in-`httpInfo`-for-custom-notifications)

Reviewer: ```suggestion
* [Additional fields in `httpInfo` for custom notifications](#additional-fields-in-httpInfo-for-custom-notifications)
```

Otherwise it seems the link doesn't work.

Author: Updated

* **throttling**: optional (must be an integer)
* **expires**: optional (must be a date or empty string "")
* **status**: optional (must be a valid status keyword)

> * **Orion extension**:

Reviewer: Maybe I didn't explain clearly enough...

What I mean is to create a new section in this .md document, I mean, this way:

```
## Additional fields in `httpInfo` for custom notifications
```

and inside that section explain the new field `includePayload`.

The section `## Subscription payload validations` doesn't need any change. Please, rollback your addition there so it looks as before.

Author: Updated

`expression` (to return only entities which have these attributes). Use them instead.

[Top](#top)

## Additional fields in `httpInfo` for custom notifications

> Orion supports a boolean field `includePayload` in `httpInfo` for custom notifications that can be used to avoid payload in the outgoing notification. If `includePayload` is set to `true` then no payload is included in the notification (and `payload` field, if used, is ignored).
> Default value if `includePayload` if omitted is `true`.

Reviewer: ```suggestion
Orion supports a boolean field `includePayload` in `httpInfo` for custom notifications that can be used to avoid payload in the outgoing notification. If `includePayload` is set to `false` then no payload is included in the notification (and `payload` field, if used, is ignored).
Default value if `includePayload` if omitted is `true` (that is, payload is included by default).
```

Note the default `true` will ensuse backward compatibility.

Author: Updated

> Default value if `includePayload` is omitted is `false`.

Reviewer: ```suggestion
Default value if `includePayload` if omitted is `false`.
```

Author: Updated

> Default value if `includePayload` if omitted is `false`.

Reviewer: I think that for backward compatibility default value should be `true` (as currently in Orion, previously to this PR, the payload is always included).

Sorry for not realizing of this until now...

`expression` (to return only entities which have these attributes). Use them instead.

[Top](#top)

## Additional fields in `httpInfo` for custom notifications

> * includePayload: Additional field in `httpInfo` for custom notifications to check not including payload or 0-length requests.
> set 'true' in the case of not including payload in the request.

Reviewer: ```suggestion
Orion supports a boolean field `includePayload` in `httpInfo` for custom notifications that can be used to avoid payload in the outgoing notification. If `includePayload` is set to `true` then no payload is included in the notification (and `payload` field, if used, is ignored).

Default value if `includePayload` is omitted is `false`.
```

Author: Updated
